staging/src/k8s.io/client-go/tools/leaderelection/leaderelection_test.go

```diff
@@ -880,6 +880,10 @@ func TestTryAcquireOrRenewConfigMapsLeases(t *testing.T) {
 }
 
 func testReleaseLease(t *testing.T, objectType string) {
+	var (
+		updates = 0
+		gets    = 0
```

Comment:
we should not do this, sharing state between test cases is not good,.
Let's wait for resolving this comment https://github.com/kubernetes/kubernetes/pull/114132/files#r1093108479 , if we Patch the test will be simpler.
If we keep using the retry we need to find another way, but I can try to help you in this case

Reply:
I agree it's bad. I haven't figured out how to optimize that,  do need some help here.

Reply:
@aojea  Friendly ping for help about the tests.

Reply:
you didn't change the implementation to use Patch instead of Update,

Reply:
> you didn't change the implementation to use Patch instead of Update,

alright, if I understand this suggestion correctly, we need to add a Patch method in the `resourcelock/interface.go:Interface`, and implement Patch for configMapLock, leaseLock and endpointLock. Then we can change the Update to Patch in `release` call. Is that correct?

Reply:
The `PatchAction` interface is quite different with `UpdateAction` and `GetAction`, I haven't figured out how to put them together in the testcases.